contracts/Killable.sol

```diff
@@ -0,0 +1,42 @@
+pragma solidity 0.5.10;
+
+import "./SafeMath.sol";
+import "./Owned.sol";
+
+contract Killable is Owned {
+    using SafeMath for uint;
+    
+    uint public constant DELAY = 30 days;
+    
+    struct KillCondition {
+        uint killingDate;
+        bool status;
+    }
+    
+    KillCondition private _killProcess;
+
+    event LogKillingProcessStarted(address emitter, uint deadline);
+    
+    modifier _whenKillingProcessNotStarted {
+        require(!_killProcess.status, "Killing process is started");
+        _;
+    }
+    
```

Comment:
Now we need to protect the user from accidentally calling this function and killing the contract which has irreversible effects; we can achieve that by turning this function into a 2-step procedure:
1. The user calls `pause()` function
2. The user calls `kill()` function: `kill()` function uses modifier `whenPaused()`
This way the user can only kill a paused contract; so we have this extra protection layer to protect them from accidental killing.